docs(router): clarify dynamic route slug is about a file name, not "folder" (#56596)
vercel/next.js: docs/03-pages/01-building-your-application/01-routing/02-dynamic-routes.mdx

## Convention

> A Dynamic Segment can be created by wrapping a filename (but not the extension) in square brackets: `[segmentName]`. For example, `[id]` or `[slug]`.

Reviewer: ```suggestion
A Dynamic Segment can be created by wrapping a file or folder name in square brackets: `[segmentName]`. For example, `[id]` or `[slug]`.
```

Author: Thanks @leerob for the review and suggestion.  I will add that I don't recall seeing an example on this page of the folder name path pattern.  Maybe that would be a good addition as well, if that feature is still intended to be used (or perhaps I've missed it elsewhere).  Thanks again!
